From e49a4fbe6f422777fad9bbd777c5745ccf8ad053 Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Tue, 26 Aug 2014 17:19:03 -0400
Subject: [input] alps: use single touch data when v3 mt data contains only one finger

Message-id: <1409073549-31548-13-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 89243
O-Subject: [PATCH RHEL7 12/18] Input: alps - use single touch data when v3 mt data contains only one finger
Bugzilla: 1107819
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1107819
Upstream: since v3.17
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7882179
Tested by me on a Toshiba Portege R30 with a v7 Alps touchpad

commit c38a448a237534aecc40bd826c55132c479f4c42
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Jul 25 22:46:53 2014 -0700

    Input: alps - use single touch data when v3 mt data contains only one finger

    For v3 protocol devices, use the more accurate single touch data when the
    mt data contains only one finger. Note the mt data reporting a finger count
    of 1 should never happen, but better safe then sorry.

    This brings the v3 bitmap handling in line with what the v4 code does,
    allowing to factor out the common bits into a helper function.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/alps.c b/drivers/input/mouse/alps.c
index 8c2de01..9a5f08d 100644
--- a/drivers/input/mouse/alps.c
+++ b/drivers/input/mouse/alps.c
@@ -669,7 +669,7 @@ static void alps_process_touchpad_packet_v3_v5(struct psmouse *psmouse)
   * If we don't have MT data or the bitmaps were empty, we have
   * to rely on ST data.
   */
- if (!fingers) {
+ if (fingers < 2) {
   f->mt[0].x = f->st.x;
   f->mt[0].y = f->st.y;
   fingers = f->pressure > 0 ? 1 : 0;
-- 
1.7.1